src/main/java/com/example/spring/controller/CartController.java

```diff
@@ -0,0 +1,68 @@
+package com.example.spring.controller;
+
+import com.example.spring.dto.cartitem.CartItemDto;
+import com.example.spring.dto.cartitem.CartItemRequestDto;
+import com.example.spring.dto.cartitem.UpdateCartItemRequestDto;
+import com.example.spring.dto.shoppingcart.ShoppingCartDto;
+import com.example.spring.service.cart.CartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springdoc.core.annotations.ParameterObject;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping Cart management", description = "Endpoints for managing shopping carts")
+@RestController
+@RequestMapping("/api/cart")
+@RequiredArgsConstructor
+public class CartController {
+    private final CartService cartService;
+
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping
+    @Operation(summary = "Add item to shopping cart",
+            description = "Add a new item to the user's shopping cart")
+    public void save(@RequestBody @Valid CartItemRequestDto cartItemRequestDto,
+                     Authentication authentication) {
+        cartService.save(authentication.getName(), cartItemRequestDto);

```

Comment:
Provide userId instead of authentication.getName() to service methods (check everywhere)